src/viam/sdk/tests/test_power_sensor.cpp

```diff
@@ -29,69 +29,33 @@ using namespace viam::sdk;
 
 BOOST_AUTO_TEST_SUITE(test_powersensor)
 
-// This sets up the following architecture
-// -- MockComponent
-//        /\
-//
-//        | (function calls)
-//
-//        \/
-// -- ComponentServer (Real)
-//        /\
-//
-//        | (grpc InProcessChannel)
-//
-//        \/
-// -- ComponentClient (Real)
-//
-// This is as close to a real setup as we can get
-// without starting another process
-//
-// The passed in lambda function has access to the ComponentClient
-//
-template <typename Lambda>
-void server_to_mock_pipeline(Lambda&& func) {
-    PowerSensorServer powersensor_server;
-    std::shared_ptr<MockPowerSensor> mock = MockPowerSensor::get_mock_powersensor();
-    powersensor_server.resource_manager()->add(std::string("mock_powersensor"), mock);
-
-    grpc::ServerBuilder builder;
-    builder.RegisterService(&powersensor_server);
-
-    std::unique_ptr<grpc::Server> server = builder.BuildAndStart();
-
-    grpc::ChannelArguments args;
-    auto grpc_channel = server->InProcessChannel(args);
-    PowerSensorClient client("mock_powersensor", grpc_channel);
-    // Run the passed test on the created stack
-    std::forward<Lambda>(func)(client, mock);
-    // shutdown afterwards
-    server->Shutdown();
-}
-
 BOOST_AUTO_TEST_CASE(test_get_voltage) {
-    server_to_mock_pipeline([](PowerSensor& client, std::shared_ptr<MockPowerSensor> mock) -> void {
+    std::shared_ptr<MockPowerSensor> mock = MockPowerSensor::get_mock_powersensor();
+    client_to_mock_pipeline<PowerSensorClient>(mock, [&](PowerSensor& client) {
         mock->peek_voltage = {10.0, false};
         BOOST_CHECK(client.get_voltage() == mock->peek_voltage);
     });
 }
 
 BOOST_AUTO_TEST_CASE(test_get_current) {
-    server_to_mock_pipeline([](PowerSensor& client, std::shared_ptr<MockPowerSensor> mock) -> void {
+    std::shared_ptr<MockPowerSensor> mock = MockPowerSensor::get_mock_powersensor();
+    client_to_mock_pipeline<PowerSensorClient>(mock, [&](PowerSensor& client) {
         mock->peek_current = {10.0, false};
         BOOST_CHECK(client.get_current() == mock->peek_current);
     });
 }
 
 BOOST_AUTO_TEST_CASE(test_get_power) {
-    server_to_mock_pipeline([](PowerSensor& client, std::shared_ptr<MockPowerSensor> mock) -> void {
+    std::shared_ptr<MockPowerSensor> mock = MockPowerSensor::get_mock_powersensor();
+    client_to_mock_pipeline<PowerSensorClient>(mock, [&](PowerSensor& client) {
         mock->peek_power = 10.0;
         BOOST_CHECK(client.get_power() == mock->peek_power);
     });
 }
 
 BOOST_AUTO_TEST_CASE(test_get_readings) {
-    server_to_mock_pipeline([](PowerSensor& client, std::shared_ptr<MockPowerSensor> mock) -> void {
+    std::shared_ptr<MockPowerSensor> mock = MockPowerSensor::get_mock_powersensor();
+    client_to_mock_pipeline<PowerSensorClient>(mock, [&](PowerSensor& client) {
```

Comment:
Don't think we need to capture by ref here.

Reply:
Done.